src/nodemetadata.cpp

```diff
@@ -101,14 +101,20 @@ void NodeMetadataList::serialize(std::ostream &os) const
 		v3s16 p = i->first;
 		NodeMetadata *data = i->second;
 
-		u16 p16 = p.Z * MAP_BLOCKSIZE * MAP_BLOCKSIZE + p.Y * MAP_BLOCKSIZE + p.X;
-		writeU16(os, p16);
-
+		if (uncompressed_pos) {
+			writeU16(os, p.X);
+			writeU16(os, p.Y);
+			writeU16(os, p.Z);
+		} else {
+			u16 p16 = p.Z * MAP_BLOCKSIZE * MAP_BLOCKSIZE + p.Y * MAP_BLOCKSIZE + p.X;
```

Comment:
i don't think this compression is useful; you only add calculs on server side & client side for 4 bytes of data